isce2_topsapp/localize_burst.py

```diff
@@ -0,0 +1,315 @@
+import io
+import re
+import copy
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterator, List, Tuple, Union
+
+import pandas as pd
+import requests
+from shapely import geometry
+
+
+@dataclass
+class BurstParams:
+    """Class that contains the parameters nessecary to request a burst from the API."""
+
+    safe_url: str
+    image_number: int
+    burst_number: int
+
+
+class BurstMetadata:
+    def __init__(self, metadata: ET.Element, manifest: ET.Element, burst_params: BurstParams):
+        self.safe_url = burst_params.safe_url
+        self.image_number = burst_params.image_number
+        self.burst_number = burst_params.burst_number
+        self.safe_name = Path(self.safe_url).with_suffix('.SAFE').name
+
+        image_numbers = [int(x.attrib['source_filename'].split('-')[-1][2]) for x in metadata]
+        products = [x.tag for x in metadata]
+        combos = list(zip(image_numbers, products))
```

Comment:
Not super in love with `combos` as I have to keep looking back to understand what's inside it. Not sure what a better name would be though...

Reply:
changed to `image_numbers_and_products`